### DISCUSSION TITLE
Any example of how to create a custom module.

### DISCUSSION THREAD

**Discussion by kevincam3** [directus/directus#6970] [Archive]
Following the documenation didn't work for me.  Directus 9.0.0-rc.85

**Comment by rijkvanzanten** [comment]
> Following the documenation didn't work for me

What didn't work?

**Comment by rijkvanzanten** [comment]
(cc @nickrum )

**Comment by kevincam3** [comment]
This didn't work:
https://docs.directus.io/guides/modules/

**Reply by rijkvanzanten** [reply]
I meant, what happened? What did not work? Did the module not show up? Did you get any errors? What's the problem?

**Comment by kevincam3** [comment]
Oh, ok. When I ran this command:
 
npx directus-extension build

It said it wasn't a directus-extension

**Reply by rijkvanzanten** [reply]
Could you share the full error?

(@nickrum)

**Comment by kevincam3** [comment]
I was inside the folder with the package.json file.

**Comment by kevincam3** [comment]
Ok, I'll recreate it now. One sec

**Comment by kevincam3** [comment]
Here's the Error message.
![image](https://user-images.githubusercontent.com/870934/127026592-73cfd9bd-cb9c-4941-8907-095321c2e183.png)

**Reply by alesvaupotic** [reply]
Hi, Kevin. Here is a short screencast on how to solve this issue. 

https://www.youtube.com/watch?v=5FHZJdVHAIk

Docs are not exactly aligned with the build process, I guess. Add 

```
    "directus:extension": {
        "type": "module",
        "path": "index.js",
        "source": "src/index.js",
        "host": "^9.0.0-rc.86",
        "hidden": false
    },

```
to package.json of your module and it should work as expected from there on. Found this info on discord:
https://discord.com/channels/725371605378924594/835244612356145167/865325002412261386

**Reply by kevincam3** [reply]
I am jumping with joy right now. Thank you soooo much. The opensource community never ceases to amaze me. I will forever be one that donates and contributes when I can.

**Reply by alesvaupotic** [reply]
You're welcome!

**Reply by rijkvanzanten** [reply]
cc @nickrum

**Comment by nickrum** [comment]
A note has been added to the docs explaining the requirement of adding a `directus:extension` field to the `package.json` file for `directus-extension build` to properly work.
